light_curves/scale_up.md

```diff
@@ -0,0 +1,558 @@
+df_lc = pd.read_parquet(parquet_dir)
+
+df_lc.sample(10)
+```
+
+Now we can make figures:
+
+```{code-cell}
+_ = create_figures(df_lc=MultiIndexDFObject(data=df_lc), show_nbr_figures=1, save_output=False)
+```
+
+## Example 3: Keyword arguments and script flags
+
++++
+
+This example shows the python `kwargs_dict` and bash script flag options in more detail.
+
+### Python `kwargs_dict`
+
+`kwargs_dict` is a dictionary containing all keyword arguments for the run. It can contain:
+- names and keyword arguments for any of the `get_*_sample` functions.
+- keyword arguments for any of the `*_get_lightcurves` functions.
+- other keyword arguments used directly by the helper.
+  These options and their defaults are shown below, further documented in the helper's `run` function.
+
+```{code-cell}
+# show kwargs_dict defaults
+helpers.scale_up.DEFAULTS
+```
+
+```{code-cell}
+# show parameter documentation
+print(helpers.scale_up.run.__doc__)
+```
+
+### Bash script flags
+
+Use the `-h` (help) flag to view the script's flag options:
+
+```{code-cell}
+# show flag documentation
+!bash code_src/helpers/scale_up.sh -h
+```
+
+### Using a yaml file
+
+It can be convenient to save the parameters in a yaml file, especially when using the bash script or in cases where you want to store parameters for later reference or re-use.
+
+Define an extended set of parameters and save it as a yaml file:
+
+```{code-cell}
+yaml_run_id = "demo-kwargs-yaml"
+
+get_samples = {
+    "green": {},
+    "ruan": {},
+    "papers_list": {
+        "paper_kwargs": [
+            {"paper_link": "2022ApJ...933...37W", "label": "Galex variable 22"},
+            {"paper_link": "2020ApJ...896...10B", "label": "Palomar variable 20"},
+        ]
+    },
+    "SDSS": {"num": 10, "zmin": 0.5, "zmax": 2, "randomize_z": True},
+    "ZTF_objectid": {"objectids": ["ZTF18aabtxvd", "ZTF18aahqkbt", "ZTF18abxftqm", "ZTF18acaqdaa"]},
+}
+
+archives = {
+    "Gaia": {"search_radius": 2 / 3600},
+    "HEASARC": {"catalog_error_radii": {"FERMIGTRIG": 1.0, "SAXGRBMGRB": 3.0}},
+    "IceCube": {"icecube_select_topN": 4, "max_search_radius": 2.0},
+    "WISE": {"radius": 1.5, "bandlist": ["W1", "W2"]},
+    "ZTF": {"nworkers": 6, "match_radius": 2 / 3600},
+}
+
+yaml_kwargs_dict = {
+    "get_samples": get_samples,
+    "consolidate_nearby_objects": False,
+    "archives": archives,
+}
+
+helpers.scale_up.write_kwargs_to_yaml(run_id=yaml_run_id, **yaml_kwargs_dict)
+```
+
+The path to the yaml file is printed in the output above.
+You can alter the contents of the file as you like.
+To use the file, set the kwarg `use_yaml=True`.
+
+Python example for the get-sample step:
+
+```{code-cell}
+sample_table = helpers.scale_up.run(build="sample", run_id=yaml_run_id, use_yaml=True)
+```
+
+Bash example:
+
++++
+
+```bash
+$ yaml_run_id=demo-kwargs-yaml
+$ bash code_src/helpers/scale_up.sh -r "$yaml_run_id" -d "use_yaml=true" -a "core"
+```
+
++++
+
+### Using a json string
```

Comment:
This is slightly too complicated for me.  Already the yaml file is a stretch, but I see that it is a good idea for bookkeeping.  Maybe leave this in the function documentation, but not in the tutorial notebook.

Reply:
Removed